File: src/main/java/fi/aalto/cs/apluscourses/intellij/generator/APlusModuleType.java
package fi.aalto.cs.apluscourses.intellij.generator;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.openapi.module.ModuleType;
import com.intellij.openapi.module.ModuleTypeManager;
import icons.PluginIcons;
import javax.swing.Icon;
import org.jetbrains.annotations.NotNull;

public class APlusModuleType extends ModuleType<APlusModuleBuilder> {

  private static final String ID = "APLUS_MODULE_TYPE";

  public APlusModuleType() {
    super(ID);
  }

  public static APlusModuleType getInstance() {
    return (APlusModuleType) ModuleTypeManager.getInstance().findByID(ID);
  }

  @Override
  public @NotNull APlusModuleBuilder createModuleBuilder() {
    return new APlusModuleBuilder();
  }

  @Override
  public @NotNull String getName() {
    return getText("intellij.ProjectBuilder.name");
  }

  @Override
  public @NotNull String getDescription() {
    return getText("intellij.ProjectBuilder.description");
  }

  @Override
  public @NotNull Icon getNodeIcon(boolean isOpened) {
    return PluginIcons.A_PLUS_LOGO_COLOR;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/generator/APlusModuleType.kt
package fi.aalto.cs.apluscourses.generator

import com.intellij.openapi.module.ModuleType
import com.intellij.openapi.module.ModuleTypeManager
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.icons.CoursesIcons
import org.jetbrains.annotations.NonNls
import javax.swing.Icon

internal class APlusModuleType : ModuleType<APlusModuleBuilder>(ID) {
    override fun createModuleBuilder(): APlusModuleBuilder {
        return APlusModuleBuilder()
    }

    override fun getName(): String {
        return MyBundle.message("aplusCourses")
    }

    override fun getDescription(): String {
        return MyBundle.message("generator.APlusModuleType.description")
    }

    override fun getNodeIcon(isOpened: Boolean): Icon {
        return CoursesIcons.LogoColor
    }

    companion object {
        @NonNls
        const val ID = "APLUS_MODULE_TYPE"

        val instance: APlusModuleType
            get() = ModuleTypeManager.getInstance().findByID(ID) as APlusModuleType
    }
}
